src/core/lib/event_engine/event_engine.cc

```diff
@@ -13,42 +13,56 @@
 // limitations under the License.
 #include <grpc/support/port_platform.h>
 
+#include <atomic>
 #include <functional>
 #include <memory>
+#include <utility>
 
 #include <grpc/event_engine/event_engine.h>
 
 #include "src/core/lib/debug/trace.h"
 #include "src/core/lib/event_engine/event_engine_factory.h"
 #include "src/core/lib/event_engine/trace.h"
-#include "src/core/lib/gprpp/sync.h"
 
 namespace grpc_event_engine {
 namespace experimental {
 
 namespace {
-const std::function<std::unique_ptr<EventEngine>()>* g_event_engine_factory =
-    nullptr;
-grpc_core::Mutex* g_mu = new grpc_core::Mutex();
+std::atomic<const std::function<std::unique_ptr<EventEngine>()>*>
+    g_event_engine_factory{nullptr};
+std::atomic<EventEngine*> g_event_engine{nullptr};
 }  // namespace
 
 void SetDefaultEventEngineFactory(
-    const std::function<std::unique_ptr<EventEngine>()>* factory) {
-  grpc_core::MutexLock lock(g_mu);
-  g_event_engine_factory = factory;
+    std::function<std::unique_ptr<EventEngine>()> factory) {
+  delete g_event_engine_factory.exchange(
+      new std::function<std::unique_ptr<EventEngine>()>(std::move(factory)));
 }
```

Comment:
Ultimately, this should maybe reset the default EventEngine, but that will be a problem given how many things rely upon GetDefaultEventEngine, and not channel args or some other engine cache (upon timer cancellation, for instance, when the original engine is not retained and the default is re-fetched).